### ISSUE TITLE
Tracking: imageops:: and DynImage::crop()/crop_imm() breakage

### ISSUE THREAD

**Issue by nabijaczleweli** [image-rs/image#1201]
Previously: https://github.com/image-rs/image/pull/1200#discussion_r409827806

1. `imageops::crop()` is to be renamed to `crop_mut()`, and `imageops::crop_imm()` to `crop()`
2. `DynImage::crop()`'s signature is to be replaced with `crop_imm()`'s, and `crop_imm()` removed

**Comment by RReverser** [comment]
Just noticed that docs of DynamicImage::crop says the signature will be replaced in 0.24 release, but looks like that never happened? Should the docs be updated? @HeroicKatora

**Comment by fintelia** [comment]
Yeah, probably should update the docs. Though if anyone is interested in making a PR, we can make note of it in #1943 to make sure they don't get forgotten when we are making the 0.25 release
